faiss/impl/NSG.cpp

```diff
@@ -285,7 +285,7 @@ void NSG::search_on_graph(
         int pool_size,
         std::vector<Neighbor>& retset,
         std::vector<Node>& fullset) const {
-    RandomGenerator gen(rand());
+    RandomGenerator gen(0x1234);
```

Comment:
Not very random if you always initialize with the same seed, or am I missing something?

Reply:
`rand()` is not thread-safe and I'm trying to make the process of searching deterministic. This change does not harm the performance according to the experiment I made.